features/shared/checkout_previous_branch_after_success/same_current_branch/previous_branch_renamed.feature

```diff
@@ -0,0 +1,11 @@
+Feature: when the previous branch is renamed during a Git Town command, the main branch becomes the new previous branch
```

Comment:
I'd say this is a bug in Git. In your example, when I was on the foo branch, then switch to the bar branch to rename foo to foo-new, and then return to my previous branch, I should end up on the foo-new branch. That seems more sane to me than going to the main branch (on which we never have been in this example).

I'm not feeling too strong about such edge cases, so I'm fine either way here. This is just another opportunity to do things better than Git, and I like that  :)